MathLib/LinAlg/Eigen/EigenMatrix.h

```diff
@@ -0,0 +1,190 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef EIGENMATRIX_H_
+#define EIGENMATRIX_H_
+
+#include <cassert>
+#ifndef NDEBUG
+#include <fstream>
+#include <string>
+#endif
+
+#include <Eigen/Sparse>
+
+#include "MathLib/LinAlg/RowColumnIndices.h"
+#include "EigenVector.h"
+
+namespace MathLib
+{
+
+/**
+ * Global matrix based on Eigen sparse matrix
+ *
+ * The matrix will be dynamically allocated during construction.
+ */
+class EigenMatrix
+{
+public:
+    typedef Eigen::SparseMatrix<double, Eigen::RowMajor> RawMatrixType;
+
+    /**
+     * constructor
+     * @param n_rows the number of rows (that is equal to the number of columns)
+     */
+    explicit EigenMatrix(std::size_t n) :_mat(n, n) {}
+
+    virtual ~EigenMatrix() {}
```

Comment:
Will we ever derive from the EigenMatrix class?
If not, the virtual specifier can be removed. Additionally I recommend to use the `final` specifier in class definition and `override` specifier for all derived functions. This can prevent some types of mistakes.